src/metabase/api/table.clj

```diff
@@ -66,14 +71,18 @@
                      :description             description))
     (api/check-500 (db/update! Table id, :visibility_type visibility_type))
     (let [updated-table (Table id)
+          driver (->> id table/table-id->database-id driver/database-id->driver) ;; is this the easy way to do this?
           new-visibility (visible-state? (:visibility_type updated-table))
           old-visibility (visible-state? original-visibility-type)
           visibility-changed? (and (not= new-visibility
```

Comment:
`visibility-changed?` should probably be renamed to `table-now-visible` or something similar.